openwisp_users/api/mixins.py

```diff
@@ -212,6 +214,90 @@ class FilterSerializerByOrgOwned(FilterSerializerByOrganization):
     _user_attr = 'organizations_owned'
 
 
+class QuerySetRequestMixin(BaseQuerySetRequestMixin):
+    def get_queryset(self, request):
+        user = request.user
+        queryset = super().get_queryset(request)
+        # superuser can see everything
+        if user.is_superuser or user.is_anonymous:
+            return queryset
+        # non superusers can see only items
+        # of organizations they're related to
+        organization_filter = getattr(user, self._user_attr)
+        # if field_name organization then just organization_filter
+        if self.field_name == 'organization':
+            return queryset.filter(pk__in=organization_filter)
+        # for field_name other than organization
+        conditions = Q(**{'organization__in': organization_filter})
+        return queryset.filter(conditions)
+
+    def __init__(self, *args, **kwargs):
+        self._user_attr = kwargs.get('user_attr')
```

Comment:
```suggestion
        self._user_attr = kwargs.pop('user_attr')
```

And remove the next line.